src/appleseed/renderer/kernel/lighting/lighttree.cpp

```diff
@@ -57,51 +60,61 @@ namespace renderer
 
 LightTree::LightTree()
   : m_tree_depth(0)
-  , m_built(false)
+  , m_is_built(false)
 {
 }
 
-LightTree::~LightTree()
-{
-    for (const_each<LightSourcePointerVector> i = m_light_sources; i; ++i)
-        delete *i;
-}
-
 bool LightTree::is_built() const
 {
-    return m_built;
+    return m_is_built;
 }
 
 size_t LightTree::build(
-    vector<NonPhysicalLightInfo>&   non_physical_lights,
-    vector<EmittingTriangle>&       emitting_triangles)
+    const std::vector<NonPhysicalLightInfo>&    non_physical_lights,
+    const std::vector<EmittingTriangle>&        emitting_triangles)
 {
+    m_non_physical_lights = non_physical_lights;
```

Comment:
Hm, here you're copying all lights into new vectors, i.e. duplicating all lights. Is that intended?